src/main/assets/js/fact-tabs/fact-tabs.js

```diff
@@ -0,0 +1,280 @@
+
+
+function FactTabs ($module) {
+  this.$module = $module;
+  this.$tabs = $module.querySelectorAll('.fact-tabs-tab');
+
+  this.keys = { left: 37, right: 39, up: 38, down: 40 };
+  this.jsHiddenClass = 'fact-tabs-panel--hidden';
+}
+
+FactTabs.prototype.init = function () {
+  if (typeof window.matchMedia === 'function') {
+    this.setupResponsiveChecks();
+  } else {
+    this.setup();
+  }
+};
+
+FactTabs.prototype.setupResponsiveChecks = function () {
+  this.mql = window.matchMedia('(min-width: 40.0625em)');
+  this.mql.addListener(this.checkMode.bind(this));
+  this.checkMode();
+};
+
+FactTabs.prototype.checkMode = function () {
+  if (this.mql.matches) {
+    this.setup();
+  } else {
+    this.teardown();
+  }
+};
+
+FactTabs.prototype.setup = function () {
+  var $module = this.$module;
+  var $tabs = this.$tabs;
+  var $tabList = $module.querySelector('.fact-tabs-list');
+  var $tabListItems = $module.querySelectorAll('.fact-tabs-list-item');
+
+  if (!$tabs || !$tabList || !$tabListItems) {
+    return;
+  }
+
+  $tabList.setAttribute('role', 'tablist');
+
+  nodeListForEach($tabListItems, function ($item) {
+    $item.setAttribute('role', 'presentation');
+  });
+
+  nodeListForEach($tabs, function ($tab) {
+    // Set HTML attributes
+    this.setAttributes($tab);
+
+    // Save bounded functions to use when removing event listeners during teardown
+    $tab.boundTabClick = this.onTabClick.bind(this);
+    $tab.boundTabKeydown = this.onTabKeydown.bind(this);
+
+    // Handle events
+    $tab.addEventListener('click', $tab.boundTabClick, true);
+    $tab.addEventListener('keydown', $tab.boundTabKeydown, true);
+
+    // Remove old active panels
+    this.hideTab($tab);
+  }.bind(this));
+
+  // Show either the active tab according to the URL's hash or the first tab
+  var $activeTab = this.getTab(window.location.hash) || this.$tabs[0];
+  this.setSelected($activeTab.innerText);
+  var selectedTab = this.$module.querySelector('.fact-tabs-title');
+  selectedTab.style.paddingTop = '0px';
+  selectedTab.style.paddingBottom= '15px';
+
+
+  this.showTab($activeTab);
+
+
+};
+
+FactTabs.prototype.teardown = function () {
+  var $module = this.$module;
+  var $tabs = this.$tabs;
+  var $tabList = $module.querySelector('.fact-tabs-list');
+  var $tabListItems = $module.querySelectorAll('.fact-tabs-list-item');
+
+  if (!$tabs || !$tabList || !$tabListItems) {
+    return;
+  }
+
+  $tabList.removeAttribute('role');
+
+  nodeListForEach($tabListItems, function ($item) {
+    $item.removeAttribute('role', 'presentation');
+  });
+
+  nodeListForEach($tabs, function ($tab) {
+    // Remove events
+    $tab.removeEventListener('click', $tab.boundTabClick, true);
+    $tab.removeEventListener('keydown', $tab.boundTabKeydown, true);
+
+    // Unset HTML attributes
+    this.unsetAttributes($tab);
+  }.bind(this));
+
+  // Remove hashchange event handler
+  window.removeEventListener('hashchange', $module.boundOnHashChange, true);
```

Comment:
The current GDS component appends a hash on to the URL and when browser history or the forward/back buttons are used, it navigates back to the appropriate tab. The hash is present in the new implementation but the forward/back button navigation is broken.

Reply:
The up and down arrows seem to working on my end tried on chrome and safari

Reply:
I'm referring to browser navigation e.g. using history, bookmarked URL or the browser's forward and back buttons. It doesn't seem to be working in Chrome or Safari for me.

Reply:
I have added forward/back navigation for browser by adding missing event handler